source/OrderedTypes/SupLattice.lagda

```diff
@@ -0,0 +1,317 @@
+
+\end{code}
+
+Now we give some naming conventions which will be useful.
+
+\begin{code}
+
+⟨_⟩ : Sup-Lattice 𝓤 𝓦 𝓥 → 𝓤  ̇
+⟨ A , rest ⟩ = A
+
+order-of : (L : Sup-Lattice 𝓤 𝓦 𝓥) → (⟨ L ⟩ → ⟨ L ⟩ → Ω 𝓦)
+order-of (A , (_≤_ , ⋁_) , rest) = _≤_
+
+syntax order-of L x y = x ≤⟨ L ⟩ y
+
+join-of : (L : Sup-Lattice 𝓤 𝓦 𝓥) → Fam 𝓥 ⟨ L ⟩ → ⟨ L ⟩
+join-of (A , (_≤_ , ⋁_) , rest) = ⋁_
+
+syntax join-of L U = ⋁⟨ L ⟩ U
+
+partial-orderedness-of : (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                       → is-partial-order ⟨ L ⟩ (order-of L)
+partial-orderedness-of (A , (_≤_ , ⋁_) , order , is-lub-of) = order
+
+reflexivity-of : (L : Sup-Lattice 𝓤 𝓦 𝓥) → is-reflexive (order-of L) holds
+reflexivity-of L = pr₁ (pr₁ (partial-orderedness-of L))
+
+antisymmetry-of : (L : Sup-Lattice 𝓤 𝓦 𝓥) → is-antisymmetric (order-of L) 
+antisymmetry-of L = pr₂ (partial-orderedness-of L)
+
+transitivity-of : (L : Sup-Lattice 𝓤 𝓦 𝓥) → is-transitive (order-of L) holds
+transitivity-of L = pr₂ (pr₁ (partial-orderedness-of L))
+
+join-is-lub-of : (L : Sup-Lattice 𝓤 𝓦 𝓥)
+               → (U : Fam 𝓥 ⟨ L ⟩)
+               → ((order-of L) Joins.is-lub-of join-of L U) U holds
+join-is-lub-of (A , (_≤_ , ⋁_) , order , suprema) = suprema
+
+join-is-upper-bound-of : (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                       → (U : Fam 𝓥 ⟨ L ⟩)
+                       → ((order-of L) Joins.is-an-upper-bound-of
+                          join-of L U) U holds
+join-is-upper-bound-of L U = pr₁ (join-is-lub-of L U)
+
+join-is-least-upper-bound-of : (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                             → (U : Fam 𝓥 ⟨ L ⟩)
+                             → ((u' , _) : Joins.upper-bound (order-of L) U)
+                             → (order-of L (join-of L U) u') holds
+join-is-least-upper-bound-of L U = pr₂ (join-is-lub-of L U)
+
+sethood-of : (L : Sup-Lattice 𝓤 𝓦 𝓥) → is-set ⟨ L ⟩
+sethood-of L =
+ type-with-prop-valued-refl-antisym-rel-is-set
+  (λ x → λ y → order-of L x y holds)
+  (λ x → λ y → holds-is-prop (order-of L x y))
+  (λ x → reflexivity-of L x)
+  (λ x → λ y → antisymmetry-of L)
+
+\end{code}
+
+We now define monotone endomaps on a sup-lattice and specify monotone endomaps
+as a special case.
+
+\begin{code}
+
+module _ {𝓤 𝓦 𝓥 : Universe} where
+
+ is-monotone : {L M : Sup-Lattice 𝓤 𝓦 𝓥} (f : ⟨ L ⟩ → ⟨ M ⟩) → 𝓤 ⊔ 𝓦  ̇
+ is-monotone {L} {M} f = (x y : ⟨ L ⟩)
+                       → (x ≤⟨ L ⟩ y) holds
+                       → (f x ≤⟨ M ⟩ f y) holds
+
+ is-monotone-endomap : {L : Sup-Lattice 𝓤 𝓦 𝓥} (f : ⟨ L ⟩ → ⟨ L ⟩) → 𝓤 ⊔ 𝓦  ̇
+ is-monotone-endomap {L} f = is-monotone {L} {L} f
+
+\end{code}
+
+We will often be taking the join of (total spaces) of subsets. It will be more
+conveinient to have some short hand for a subset and the corresponding family
+that consists of its total space and inclusion map.
+
+\begin{code}
+
+module _
+        {𝓤 𝓦 𝓥 : Universe}
+        (L : Sup-Lattice 𝓤 𝓦 𝓥)
+        {A : 𝓥  ̇}
+        (m : A → ⟨ L ⟩)
+       where
+
+ subset-to-family : 𝓟 {𝓥} A → Fam 𝓥 ⟨ L ⟩
```

Comment:
This definition should not depend on `L` being a sup-lattice. When it *is* a sup-lattice, you could introduce the notation `⋁ₛ⟨ L ⟩ A` where `A` is some subset to avoid writing `subset-to-family` all the type.

Reply:
Actually, `subset-to-family` should probably be moved to the file defining families or the file defining (generalized) powersets (whichever makes more sense import-wise).

I would also make two versions, one which turns subsets of `L` into families, and one like you have now which generalizes this (it is a generalization by taking `m = id`).

Reply:
I added subset-to-family in the Slice.Families file. I felt this was more appropriate. Althoug I'm having a bit of trouble with the syntax, so I've commented it out. Will take a look at my recent push and see if you can give me some pointers. I still don't have a good hold on what is or isn't allow with syntax. @tomdjong

Reply:
Alright, I figured out why my syntax wasn't working. Let me know your thoughts.